src/com/goide/inspections/GoPlaceholderCountInspection.java

```diff
@@ -16,170 +16,256 @@
+      String message = "Value used for formatting text does not appear to contain a placeholder";
+      holder.registerProblem(placeholder, message, ProblemHighlightType.GENERIC_ERROR_OR_WARNING);
+      return;
     }
-    if (resolved instanceof GoConstDefinition) {
-      return getValue(((GoConstDefinition)resolved).getValue());
+
+    callArgsNum--;
+    List<Placeholder> placeholders = GoPlaceholderChecker.parsePrintf(placeholderText);
+    for (Placeholder fmtPlaceholder : placeholders) {
+      if (!checkPrintfArgument(holder, placeholder, callExpr, arguments, callArgsNum, placeholderPosition, fmtPlaceholder)) return;
     }
 
-    return null;
-  }
+    if (hasErrors(holder, placeholder, placeholders)) return;
+    // TODO florin add a min argument to see if we are skipping any argument from the formatting string
+    int maxArgsNum = computeMaxArgsNum(placeholders, placeholderPosition);
 
-  // todo: implement ConstEvaluator
-  @Nullable
-  private static String getValue(@Nullable GoExpression expression) {
-    if (expression instanceof GoStringLiteral) {
-      return expression.getText();
+    if (GoPsiImplUtil.hasVariadic(callExpr.getArgumentList()) && maxArgsNum >= callArgsNum) {
+      return;
     }
-    if (expression instanceof GoAddExpr) {
-      String sum = getValue((GoAddExpr)expression);
-      return sum.isEmpty() ? null : sum;
+
+    if (maxArgsNum != callArgsNum) {
+      int expect = maxArgsNum - placeholderPosition;
+      int numArgs = callArgsNum - placeholderPosition;
+      String message = String.format("Got %d placeholder(s) for %d arguments(s)", expect, numArgs);
+      holder.registerProblem(placeholder, message, ProblemHighlightType.GENERIC_ERROR_OR_WARNING);
     }
 
-    return null;
+    // TODO florin: check if all arguments are strings and offer to replace with Println and string concat
   }
 
-  // todo: implement ConstEvaluator
-  @NotNull
-  private static String getValue(@Nullable GoAddExpr expression) {
-    if (expression == null) return "";
-    StringBuilder result = new StringBuilder();
-    for (GoExpression expr : expression.getExpressionList()) {
-      if (expr instanceof GoStringLiteral) {
-        result.append(expr.getText());
+  private static int computeMaxArgsNum(@NotNull List<Placeholder> placeholders, int firstArg) {
+    int maxArgsNum = 0;
+    for (Placeholder placeholder : placeholders) {
+      List<Integer> arguments = placeholder.getArguments();
+      if (arguments.isEmpty()) continue;
+      int max = Collections.max(arguments);
+      if (maxArgsNum < max) {
+        maxArgsNum = max;
+      }
+    }
+
+    return maxArgsNum + firstArg;
+  }
+
+  private static boolean hasErrors(@NotNull ProblemsHolder holder,
+                                   @NotNull GoExpression placeholder,
+                                   @NotNull List<Placeholder> placeholders) {
+
+    for (Placeholder p : placeholders) {
```

Comment:
it's really hard to understand why while traversing `placeholders` you have `placeholder` variable which is not instance of `Placeholder` class. Unreadable, please rename it properly